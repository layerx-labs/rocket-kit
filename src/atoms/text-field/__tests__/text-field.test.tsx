import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { TextField } from '../../..';

describe('TextField', () => {
  it('renders', () => {
    const { asFragment } = render(
      <TextField type="text" value="this is my text" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct value', () => {
    const textValue = 'this is my text';
    render(<TextField type="text" value={textValue} />);
    const foundInput = screen.getByDisplayValue(textValue);
    expect(foundInput).toHaveValue(textValue);
  });

  it('has empty value', () => {
    render(<TextField type="text" />);
    const foundInput = screen.getByRole('textbox');
    expect(foundInput).toHaveValue('');
  });

  it('has correct placeholder', () => {
    const placeholderValue = 'this is my placeholder';
    render(<TextField type="text" placeholder={placeholderValue} />);
    const foundInput = screen.getByPlaceholderText(placeholderValue);
    expect(foundInput).toHaveProperty('placeholder', placeholderValue);
  });

  it('has empty placeholder', () => {
    render(<TextField type="text" />);
    const foundInput = screen.getByRole('textbox');
    expect(foundInput).toHaveProperty('placeholder', '');
  });

  it('has correct value after typing', () => {
    const textValue = 'this is my text';
    render(<TextField type="text" />);
    const foundInput = screen.getByRole('textbox');
    userEvent.type(foundInput, textValue);
    expect(foundInput).toHaveValue(textValue);
  });

  it('has correct date value', () => {
    const dateValue = '2021-01-19';
    render(<TextField type="date" value={dateValue} />);
    const foundInput = screen.getByDisplayValue(dateValue);
    expect(foundInput).toHaveValue(dateValue);
  });
});
