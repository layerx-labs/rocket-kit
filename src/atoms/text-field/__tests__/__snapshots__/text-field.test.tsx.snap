// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders 1`] = `
<DocumentFragment>
  .c0 {
  border-width: 0 0 0.0625rem 0;
  border-style: solid;
  border-color: #939393;
  border-radius: 0;
  height: 2.5rem;
  padding: 0;
  font-family: inherit;
  font-size: 1rem;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0:focus {
  border-color: #4329a6;
  outline: none;
}

.c0:disabled {
  background-color: #c4c7d3;
  color: #939393;
}

.c0:invalid {
  border-color: #ef5766;
  outline: none;
  box-shadow: none;
  -webkit-box-shadow: none;
  -moz-box-shadow: none;
}

.c0:required {
  border-color: #939393;
}

.c0::-webkit-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c0::-moz-placeholder {
  color: #939393;
  opacity: 1;
}

.c0:-ms-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c0::placeholder {
  color: #939393;
  opacity: 1;
}

.c0:-ms-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c0::-ms-input-placeholder {
  color: #939393;
  opacity: 1;
}

<input
    class="c0 text-field"
    placeholder=""
    type="text"
    value="this is my text"
  />
</DocumentFragment>
`;
